Wangscape/noise/Reseedable.h

```diff
@@ -58,8 +58,8 @@ struct Reseedable final
 template<typename InputIt>
 inline Reseedable Reseedable::terrace(InputIt first, InputIt last, bool inverted)
 {
-    static_assert(std::is_same<std::iterator_traits<InputIt>::iterator_category::input_iterator_tag,
-                  std::input_iterator_tag>::value,
+    static_assert(std::is_same<typename std::iterator_traits<InputIt>::iterator_category,
```

Comment:
Does this still compile if you instantiate the template with an `InputIt` which is not an InputIterator, but is something more powerful, such as BidirectionalIterator? There should probably be a test to make sure the intended cases compile.